### PR TITLE
Ac sk contact us

### LINKED ISSUE #34: Contact Page
Steps:

- [x] Implement Contact us page according to designs
- [ ] Send an email to the partners every time someone submits the form with the form answers (optional)

### PR DESCRIPTION
**Pull request by al-chang**
### ℹ️ Issue

Closes #34 

### 📝 Description

- Built the contact us page
- Also refactored pop up component to use react children for body of popup because I wanted to do that before but forgot and then the pr with the component got merged 

### ✔️ Verification
 
Screen Shot 2022-05-26 at 10 10 12 PM

### 🏕️ (Optional) Future Work / Notes

- Need to add functionality for form to register people for mailing list. Currently the form exists in component but we are not rendering it. This would involve figuring out what mailing service we would use to actually handle that.

### CODE REVIEW

**Comment by rymaju** [comment]
Orthogonal changes belong in separate PRs! Please consider splitting the work up next time.

File: lib/api/emailLogic.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const signUpForMailingList = () => {
+  /* TOOD */
```
Comment (rymaju) [inline lib/api/emailLogic.ts:2]:
```suggestion
  /* TODO */
```
⛏️

File: pages/api/email/sign-up.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * @swagger
+ * /api/email/sign-up:
+ *   post:
+ *     description: Send email to partner to confirm sign-up
+ *     parameters:
+ *       - name: firstName
+ *         required: true
+ *         description: First name of individual.
```
Comment (rymaju) [inline pages/api/email/sign-up.ts:13]: ❓ Do these really need descriptions if each on is just `<the thing> of individual`?

File: pages/contact/index.tsx [hunk 3]
```diff
@@ -1,5 +1,107 @@
...
+            <Text>
+              For questions, press inquiries, or other issues please contact us at{' '}
+              <b>
+                <a href="mailto:info@outstandinglife.org">info@outstandinglife.org</a>
+              </b>
+              .
+            </Text>
+          </Box>
+          {/* hiding form for now because we don't have registration functionality */}
+          {false && (
```
Comment (rymaju) [inline pages/contact/index.tsx:36]: 💭 This is great! Hiding unfinished code behind a false value is similar to "feature flagging", and we ought to do it here!

If you had a `featureFlags.ts` file, exporting a boolean, you could change this line to:
```
CONTACT_FORM_FLAG && ...
```

**Comment by al-chang** [comment]
yeah my bad, forgot to do it before and it was like two lines so figured it was no big but ill split it up next time @rymaju

File: pages/contact/index.tsx (hunk 3 above)
Comment (rymaju) [reply pages/contact/index.tsx:36]: ⛏️ Forms ought to be their own component

File: pages/donate/index.tsx [hunk 4]
```diff
@@ -7,12 +7,16 @@ const Donate: React.FC = () => {
...
         header="Open different page"
-        body="This donation button will take you to MAC’s donation page where you will find a YELLOW button that says OUTstandingLIFE. All donations go through their secure account and are then credited to OUTstandingLIFE."
         manageModal={modalManagement}
         onConfirm={() => {
           window.open('https://www.mac-boston.org/donate', '_blank');
-        }}
-      />
+        }}>
+        <Text>
+          This donation button will take you to MAC’s donation page where you will find a YELLOW
```
Comment (rymaju) [inline pages/donate/index.tsx:15]: 💭 Would it be good to show images or a gif displaying exactly what to press rather than describing the UI via text?

**Review by rymaju — APPROVED** [review]

File: pages/contact/index.tsx (hunk 3 above)
Comment (varunthakkar1) [reply pages/contact/index.tsx:36]: agreed with both

File: pages/donate/index.tsx (hunk 4 above)
Comment (varunthakkar1) [reply pages/donate/index.tsx:15]: This is the text that they gave us lol

File: pages/api/email/sign-up.ts (hunk 2 above)
Comment (al-chang) [reply pages/api/email/sign-up.ts:13]: Prob not but that's more so just placeholder text until the actual route logic is built

File: pages/contact/index.tsx (hunk 3 above)
Comment (saahilkumar) [reply pages/contact/index.tsx]: Addressed both of these comments :​)

**Review by varunthakkar1 — APPROVED** [review]

**Pull request merged**